### DISCUSSION TITLE
Server admin to recover deleted users' device groups

### DISCUSSION THREAD

**Discussion by D4V3M0NK** [Ylianst/MeshCentral#5530] [Issues]
I am a full server admin on my MC2 system. A user had created a couple of device groups and added remotes into them. That user was deleted from the system and now, even as full server admin, I can't see the device groups they created.

Is there any way of recovering these remotes?

I do have my user account set up as `orphanAgentUser` and I'm using MySQL as backend DB.

**Comment by D4V3M0NK** [comment]
Further investigation shows that there's an entry in the `main` table, type of `mesh` and appears to show the group being set up. The `creatorid` and `creatorname` is set to the deleted users' account. Can I simply update that entry, assigning my id/name?

**Comment by si458** [comment]
Sadly no, once a user is DELETED its DELETED!

we use the DELETE FROM query for mysql, so unless u have a backup of ur db. The isn't much u can do

**Comment by D4V3M0NK** [comment]
Ok - that I understand ... once the user is deleted, they're deleted. No issue with that ... however I would imagine (?) that the remotes aren't removed from the server themselves, just the user is removed from the UI. Assuming that, the remotes are therefore still running and connecting to the server - therefore I would expect if the group hash that they can't connect to isn't available, the `orphanAgentUser` comes into play and a default group is created, and those remotes placed into that group, correct?

**Comment by si458** [comment]
In theory u r correct, that's the idea of orphanAgentUser, but ive never tired it

**Comment by D4V3M0NK** [comment]
@si458 : I believe I've discovered a little bug that's related to this...

A remote is moved from ServerA to User2's group on ServerB ... and then User2 is deleted, the remote disappears too (as we've discovered) because it was joined to User2's group and now that that user has been deleted, the remote is removed from the UI*.

Through client interaction, we are able to move the remote back to ServerA so that we can move it to ServerB again, but have it join User1's group. However, it never appears on ServerB.

My theory is that there's still a record of the remote on ServerB, so that when it reconnects again, it joins the group that it was initially registered to, and of course, that was User2's group - which is no longer available, because User2 has been deleted.

Therefore, my question : is there a way of manipulating the database to totally remove the remote's entry from ServerB? If that were possible, when we try to move it from ServerA to ServerB, it will join the group it was intended for, and not use the "old" entry in the database and join it's original group (which of course, probably exists but isn't visible)

*I believe this is where the issue lies - at this point, the remote is orphaned - however the MC2 code doesn't appear to check this and therefore the `orphanAgentUser` never gets assigned.

Thoughts?
